src/flexmeasures_client/client.py

```diff
@@ -0,0 +1,287 @@
+
+    async def post_measurements(
+        self,
+        sensor_id: int,
+        start: str,
+        duration: str,
+        values: list[float],
+        unit: str,
+        entity_address: str,
+        prior: str | None = None,
+    ):
+        """Post sensor data for the given time range."""
+        json = dict(
+            sensor=f"{entity_address}.{sensor_id}",
+            start=pd.Timestamp(
+                start
+            ).isoformat(),  # for example: 2021-10-13T00:00+02:00
+            duration=pd.Timedelta(duration).isoformat(),  # for example: PT1H
+            values=values,
+            unit=unit,
+        )
+        if prior:
+            json["prior"] = prior
+
+        response, status = await self.request(
+            uri="sensors/data",
+            json=json,
+        )
+        check_for_status(status, 200)
+        print("Sensor data sent successfully.")
+
+    async def post_schedule_trigger(
+        self,
+        sensor_id: int,
+        start: str,
+        soc_unit: str,
+        soc_at_start: float,
+        soc_targets: list,
+        consumption_price_sensor: int | None = None,
+        production_price_sensor: int | None = None,
+        inflexible_device_sensors: list[int] | None = None,
+    ):
+        """Post schedule trigger with initial and target states of charge (soc)."""
+        message = {
+            "start": pd.Timestamp(
+                start
+            ).isoformat(),  # for example: 2021-10-13T00:00+02:00
+            "flex-model": {
+                "soc-unit": soc_unit,
+                "soc-at-start": soc_at_start,
+                "soc-targets": soc_targets,
+            },
+            "flex-context": {},
+        }
+
+        # Set optional flex context
+        if consumption_price_sensor is not None:
+            message["flex-context"][
+                "consumption-price-sensor"
+            ] = consumption_price_sensor
+        if production_price_sensor is not None:
+            message["flex-context"]["production-price-sensor"] = production_price_sensor
+        if inflexible_device_sensors is not None:
+            message["flex-context"][
+                "inflexible-device-sensors"
+            ] = inflexible_device_sensors
+
+        response, status = await self.request(
+            uri=f"sensors/{sensor_id}/schedules/trigger",
+            json=message,
+        )
+        check_for_status(status, 200)
+        print("Schedule triggered successfully.")
+
+    async def get_schedule(
+        self,
+        sensor_id: int,
+        schedule_id: str,
+        duration: str,
+    ):
+        """Get schedule with given ID."""
+        response, status = await self.request(
+            uri=f"sensors/{sensor_id}/schedules/{schedule_id}",
+            method="GET",
+            json={
+                "duration": pd.Timedelta(duration).isoformat(),  # for example: PT1H
+            },
+        )
+        check_for_status(status, 200)
```

Comment:
More of  general comment: I think we should create some `Exception`s to handle errors in the request and raise them when needed. That way, we can implement custom rollback strategies in the code that uses the client.

Surely, this should be discussed in a separate thread.

Reply:
Created an issue: https://github.com/FlexMeasures/flexmeasures-client/issues/13

Not sure what the should be in the issue exactly so the issue is still pretty vague. I'm a bit new to exception handling so might be something we can improve in general.